core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -157,6 +156,12 @@ abstract class AbstractFetcherThread(name: String,
                     fetcherLagStats.getAndMaybePut(topic, partitionId).lag = Math.max(0L, partitionData.highWatermark - newOffset)
                     // Once we hand off the partition data to the subclass, we can't mess with it any more in this thread
                     processPartitionData(topicPartition, currentPartitionFetchState.offset, partitionData)
+
+                    if (validBytes > 0) {
+                      // Update partitionStates only if there is no exception during processPartitionData
+                      partitionStates.updateAndMoveToEnd(topicPartition, new PartitionFetchState(newOffset))
+                      fetcherStats.byteRate.mark(validBytes)
+                    }
```

Comment:
If the partition has a corrupted message. The fetcher thread will keep fetching the same message set. In this case, it seems that we should have some back off in order to avoid getting exception in every fetch request. Maybe we can just use ```replica.fetch.backoff.ms``` but only apply to that partition.